src/components/GenericAccount.tsx

```diff
@@ -88,16 +89,29 @@ const GenericAccount = ({ value }: Props) => {
     dispatchTransaction(TransactionActionCreators.setReceiverAddress(derivedAddress));
   };
 
+  const shortGenericAddress = shorterItem(value);
   return (
     <Container className={classes.container}>
       {(!selected || selected === NATIVE) && (
         <div className={classes.native} onClick={setNativeAsTarget}>
-          <AccountDisplay accountName="Native" address={nativeAddress} balance={nativeState.formattedBalance} />
+          <AccountDisplay
+            address={nativeAddress}
+            addressKind="native"
```

Comment:
if it is a kind, it is not better to move this to a constant?

Reply:
even an enum ....

Reply:
Well, I thought that the reason for having TypeScript is to make sure this kind of stuff is checked.

If I define the type to be `'native' | 'companion'` then I can't make a typo anyway. Enums obviously are super cool if there amount of values is much greater, so that you don't have to repeat the type, but they also introduce boilerplate.

I'm totally fine with an enum though, sure, will update.